Makefile

```diff
@@ -3,21 +3,56 @@ SHELL := /bin/bash # Use bash syntax
 # Set up variables
 GO111MODULE=on
 
-# Build ldflags
-VERSION ?= "v0.0.0"
-GITCOMMIT=$(shell git rev-parse HEAD)
-BUILDDATE=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
-GO_LDFLAGS=-ldflags "-X main.version=$(VERSION) \
-			-X main.buildHash=$(GITCOMMIT) \
-			-X main.buildDate=$(BUILDDATE)"
+GO_CMD_FLAGS=-tags codegen
+AWS_SERVICE=$(shell echo $(SERVICE))
+SERVICE_MODEL_NAME=$(shell echo $(MODEL_NAME))
+ifeq ($(SERVICE_MODEL_NAME),)
+  SERVICE_MODEL_NAME:=""
+endif
+ROOT_DIR=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+CONTROLLER_BOOTSTRAP=./bin/controller-bootstrap
+CONTROLLER_DIR=${ROOT_DIR}/../${AWS_SERVICE}-controller
+CODE_GEN_DIR=${ROOT_DIR}/../code-generator
+AWS_SDK_GO_VERSION=$(shell curl -H "Accept: application/vnd.github.v3+json" \
+                                 https://api.github.com/repos/aws/aws-sdk-go/releases/latest | jq -r '.tag_name')
+ACK_RUNTIME_VERSION=$(shell curl -H "Accept: application/vnd.github.v3+json" \
+								 https://api.github.com/repos/aws-controllers-k8s/runtime/releases/latest | jq -r '.tag_name')
+.DEFAULT_GOAL=run
+EXISTING_CONTROLLER="true"
+DRY_RUN="false"
 
-.PHONY: all test
+.PHONY: build, generate, update, init, run, clean
 
-all: test
+build:
+	@go build ${GO_CMD_FLAGS} -o ${CONTROLLER_BOOTSTRAP} ./cmd/controller-bootstrap/*.go
 
-test: 				## Run code tests
-	go test -v ./...
+generate: build
+	@${CONTROLLER_BOOTSTRAP} generate -s ${AWS_SERVICE} -r ${ACK_RUNTIME_VERSION} -v ${AWS_SDK_GO_VERSION} -d=${DRY_RUN} -o ${ROOT_DIR}/../${AWS_SERVICE}-controller -m ${SERVICE_MODEL_NAME}
 
-help:           	## Show this help.
-	@grep -F -h "##" $(MAKEFILE_LIST) | grep -F -v grep | sed -e 's/\\$$//' \
-		| awk -F'[:#]' '{print $$1 = sprintf("%-30s", $$1), $$4}'
\ No newline at end of file
+update: build
+	@${CONTROLLER_BOOTSTRAP} generate -s ${AWS_SERVICE} -r ${ACK_RUNTIME_VERSION} -v ${AWS_SDK_GO_VERSION} -d=${DRY_RUN} -e=${EXISTING_CONTROLLER} -o ${ROOT_DIR}/../${AWS_SERVICE}-controller -m ${SERVICE_MODEL_NAME}
+
+init: generate
+	@export SERVICE=${AWS_SERVICE}
+	@echo "build controller attempt #1"
```

Comment:
These echos give off the feeling that the process is actually not working properly. `attempt` sounds like it is retrying from a failure and `missing go.sum entry` might confuse someone who thinks they need to provide it. I actually think it would be fine if we don't print anything and just let it run - we can print an error if something fails or a final message at the end saying if it succeeded?

Reply:
These echos have been removed and added a final message at the end if it succeeded.